lingua_franca/lang/parse_en.py

```diff
@@ -460,7 +461,12 @@ def _extract_whole_number_with_text_en(tokens, short_scale, ordinals):
             # six hundred sixty six
             # two million five hundred thousand
             if word in multiplies and next_word not in multiplies:
-                if all([string_num_scale.get(other_token.word) < current_val if other_token.word in multiplies else True for other_token in tokens[idx::]]):
+                # iterate over the remaining words, and check for larger powers
+                # of ten. if we find one, skip this part, preserving the current
+                # value for the next iteration
+                if all([string_num_scale.get(other_token.word) < current_val
+                        if other_token.word in multiplies else True
+                        for other_token in tokens[idx+1:]]):
```

Comment:
The ternary expression tripped me up for a while here :) maybe `or other_token.word not in multiplies` could be considered?

The ternary expression inside the list comprehension made the logic tricky to grasp for me.

Also is the `.get` adding anything here? if the word isn't in `string_num_scale` it will return `None` and the comparison to `current_val` would fail.

Reply:
`multiplies` and `string_num_scale` derive from the same collection. `multiplies` should be `string_num_scale.values`.

In this case, all the `get()` was really changing was the exception it'd raise, in case we break one of these functions. `get()` would raise a TypeError, index lookup a KeyError. On reflection, the KeyError's clearer, so, good call.

I like the `all()` call because it's an arbitrary number of comparisons. If it *is* time to append
`val` to `to_sum`, we're going to iterate over the entire string from the current word forward. `else True` allows it to work without additional comparisons, while ensuring that we'll stop iterating as soon as we hit one `False`.

OTOH, I recognize that this is confusing, so I've written a VERY long inline explanation.

Reply:
I suppose it could be a filter, that might be more readable.

Reply:
Turns out it's better as a for loop.

Reply:
I didn't have an issue with the all(), I thought that was quite clear, it was the ternary expression in the list comprehension that tripped me up and I thought an `or` might be clearer. As I said it was a personal preference

Reply:
I figured, but after staring at it some more, I realized that a straight for loop is much, *much* more efficient here, as the all() arrangement still iterated over the whole string, every time around. This loop stops as soon as it hits False, like all() usually would.

Reply:
Ah, good call.